### ISSUE TITLE
Op request: Tensorflow ScatterNd node

### ISSUE THREAD

**Issue by Max-K-cadence** [onnx/onnx#2074]
TensorFlow has a node ScatterNd which is described here: https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/scatter-nd

The Numpy implementation of this functionality is trivial:
```
  def reference_impl(indices, updates, shape):
    data = np.zeros(shape).astype(np.float32)
    for i, u in zip(indices, updates):
      data[tuple(i)] += u
    return data
```
However, I was not able to find any reasonably easy way to express this in terms of ONNX nodes.  The closest analogue is the Scatter node, but it does not support index duplication, and it cannot be used to make multi-dimensional updates without sophisticated tensor manipulations before calling the Scatter.

Basically the ScatterNd operation is a generalized version of "add value by arbitrary index" operation. Would that be useful to have such operation in ONNX?

**Comment by wschin** [comment]
If we change `Scatter` to do `+=` instead of `=`, would this problem be solved?

**Comment by Max-K-cadence** [comment]
Hi @wschin,

I believe it wouldn't. ScatterNd is a completely different node. For example, ONNX Scatter always expects all Input, Indices and Updates to have the same rank, while ScatterNd allows scattering 3d values by 2d indices into 5d tensor. Formula for ScatterNd:

```
  def reference_impl(indices, updates, shape):
    data = np.zeros(shape).astype(np.float32)
    for i, u in zip(indices, updates):
      data[tuple(i)] += u
    return data
```

For Scatter, something like
```
  def reference_impl(indices, updates, shape):
    data = np.zeros(shape).astype(np.float32)
    for i ...:
      for j ...:
        for k...:
          data[indices[i][j][k]][j][k] = u
    return data
```
So may not similar names confuse you, this is a completely different node.

**Comment by wschin** [comment]
They have different behaviors but as we can convert `ScatterNd`'s indices/updates to `Scatter`'s indices/updates. With the transformed indices and updates, `Scatter` is able to produce `ScatterNd`'s output. Is my understanding correct?

**Comment by Max-K-cadence** [comment]
If we convert indices - yes. The problem is that the generalized conversion of `ScatterNd` indices to `Scatter` indices is equivalent to `ScatterNd` operation itself. :)

**Comment by BowenBao** [comment]
I have opened a PR proposing `ScatterND` to ONNX https://github.com/onnx/onnx/pull/2220

**Comment by jcwchen** [comment]
Since ScatterND has been supported, close this issue now. Thanks @BowenBao for the contribution!
